database/misc/redis.py

```diff
@@ -257,15 +253,12 @@ def main():
 
     # flush Command section -----------
     elif command == "flush":
-        try:
-            db = int(module.params['db'])
-        except Exception:
-            db = 0
+        db = module.params['db']
         mode = module.params['flush_mode']
 
         #Check if we have all the data
         if mode == "db":
-            if type(db) != int:
+            if db is None:
                 module.fail_json(
```

Comment:
Okay..  well, this PR fixes things, the module is in extras, and this note isn't a bug... It's just not making best use of the arg parsing framework.  I'm going to merge this PR and if the module owner is inclined, they look at making this change as well.